WordPressKit/Jetpack Scan/JetpackScanThreat.swift

```diff
@@ -0,0 +1,294 @@
+            context = JetpackThreatContext(with: contextDict)
+        } else if ((try container.decodeIfPresent(String.self, forKey: .context)) != nil) {
+            context = JetpackThreatContext.emptyObject()
+        } else {
+            context = nil
+        }
+
+        // Calculate the type of threat last
+        type = ThreatType(threat: self)
+    }
+
+    private enum CodingKeys: String, CodingKey {
+        case id
+        case signature
+        case description
+        case fileName = "filename"
+        case firstDetected = "first_detected"
+        case fixable
+        case `extension`
+        case diff
+        case context
+        case rows
+        case status
+        case fixedOn = "fixed_on"
+    }
+}
+
+/// An object that describes how a threat can be fixed
+public struct JetpackScanThreatFixer: Decodable {
+    public enum ThreatFixType: String {
+        case replace
+        case delete
+        case update
+        case edit
+
+        case unknown
+    }
+
+    /// The suggested threat fix type
+    public let type: ThreatFixType
+
+    /// The file path of the file to be fixed
+    public var file: String? = nil
+
+    /// The target version to fix to
+    public var target: String? = nil
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        let typeString = try container.decode(String.self, forKey: .type)
+        type = ThreatFixType(rawValue: typeString) ?? .unknown
+
+        file = try container.decodeIfPresent(String.self, forKey: .file)
+        target = try container.decodeIfPresent(String.self, forKey: .target)
+    }
+
+    private enum CodingKeys: String, CodingKey {
+        case type = "fixer"
+        case file
+        case target
+    }
+}
+
+/// Represents plugin or theme additional metadata
+public struct JetpackThreatExtension: Decodable {
+    public enum JetpackThreatExtensionType: String {
+        case plugin
+        case theme
+
+        case unknown
+    }
+
+    public let slug: String
+    public let name: String
+    public let type: JetpackThreatExtensionType
+    public let isPremium: Bool
+    public let version: String
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        slug = try container.decode(String.self, forKey: .slug)
+        name = try container.decode(String.self, forKey: .name)
+        version = try container.decode(String.self, forKey: .version)
+        isPremium = try container.decode(Bool.self, forKey: .isPremium)
+        
+        let typeString = try container.decode(String.self, forKey: .type)
+        type = JetpackThreatExtensionType(rawValue: typeString) ?? .unknown
+    }
+
+    private enum CodingKeys: String, CodingKey {
+        case type
+        case slug
+        case name
+        case version
+        case isPremium
+    }
+}
```

Comment:
```suggestion
/// Represents plugin or theme additional metadata
public struct JetpackThreatExtension: Decodable {
    public enum JetpackThreatExtensionType: String, Decodable {
        case plugin
        case theme

        case unknown
    }

    public let slug: String
    public let name: String
    public let type: JetpackThreatExtensionType
    public let isPremium: Bool
    public let version: String
}
```

This same logic can be used to simplify other entities I guess.